File: mobile/SmartSolarMobile/app/src/main/java/com/smartsolar/mobile/data/remote/dto/AvailableSlotResponse.java
package com.smartsolar.mobile.data.remote.dto;

public final class AvailableSlotResponse {
    private String slotId;
    private String stationId;
    private String startAtUtc;
    private String endAtUtc;
    private int availableSlots;
    private int totalSlots;

    public AvailableSlotResponse() { }

    public AvailableSlotResponse(String slotId, String stationId, String startAtUtc, String endAtUtc, int availableSlots, int totalSlots) {
        this.slotId = slotId;
        this.stationId = stationId;
        this.startAtUtc = startAtUtc;
        this.endAtUtc = endAtUtc;
        this.availableSlots = availableSlots;
        this.totalSlots = totalSlots;
    }

    public String getSlotId() { return slotId; }
    public String getStationId() { return stationId; }
    public String getStartAtUtc() { return startAtUtc; }
    public String getEndAtUtc() { return endAtUtc; }
    public int getAvailableSlots() { return availableSlots; }
    public int getTotalSlots() { return totalSlots; }
}

File: src/SmartSolar.Application/DTOs/Reservations/AvailableSlotResponse.cs
/*
 * File: AvailableSlotResponse.cs
 * Project: Smart Solar Microgrid Trading System
 * Purpose: Exposes active booking slot summary for assisted/self-service slot selection.
 * Note: Keep this header and update method-level comments as the code evolves.
 */
namespace SmartSolar.Application.DTOs.Reservations;

public sealed record AvailableSlotResponse(
    string SlotId,
    string StationId,
    DateTime StartAtUtc,
    DateTime EndAtUtc,
    int AvailableSlots,
    int TotalSlots);
